genesis/genesis.go

```diff
@@ -95,3 +95,10 @@ func (g *Genesis) Commit(ctx context.Context, app monomer.Application, blockStor
 	}
 	return nil
 }
+
+func (g *Genesis) predeployContracts(ethState *state.StateDB) *state.StateDB {
+	for _, predeploy := range contracts.Predeploys {
+		ethState.SetCode(predeploy.Address, predeploy.DeployedBytecode)
```

Comment:
Optimism does something suprisingly similar for genesis, however it's in a much less direct way. They use a [foundry contract script](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/L2Genesis.s.sol#L47) to create a full state dump for genesis. In that script, each predeployed contract address uses a helper function to [set its contract code](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/L2Genesis.s.sol#L519-L520) to the expected deployed bytecode. This state dump is then used to [directly set the optimism genesis state](https://github.com/ethereum-optimism/optimism/blob/develop/op-chain-ops/genesis/layer_two.go#L43).

Since our current predeployed contracts have no constructors, all we need to do is set the code for the contract account. However, if we eventually need to support predeployed contracts with constructors, we may need to support setting storage or balance as well.

Reply:
Ok. Let's add a todo or task on the board for this. Foundry deploy system would be nice but overkill right now.